src/hotspot/share/gc/shared/oopStorage.hpp

```diff
@@ -107,9 +107,25 @@ class OopStorage {
 
   // Allocates and returns a new entry.  Returns NULL if memory allocation
   // failed.  Locks _allocation_mutex.
-  // postcondition: *result == NULL.
+  // postcondition: result == NULL or *result == NULL.
   oop* allocate();
 
+  // Maximum number of entries that can be obtained by one call to
+  // allocate(oop**, size_t).
+  static const size_t bulk_allocate_limit = BitsPerWord;
+
+  // Allocates multiple entries, returning them in the ptrs buffer.  The
+  // number of entries that will be allocated is never more than the minimum
+  // of size and bulk_allocate_limit, but may be less than either.  Possibly
```

Comment:
It took me a few seconds to understand this sentence. Maybe replace it with "postcondition: result <= min(size, bulk_allocate_limit)" below?

Reply:
done